crates/hc_sandbox/src/config.rs

```diff
@@ -9,16 +9,23 @@ pub const CONDUCTOR_CONFIG: &str = "conductor-config.yaml";
 
 /// Create a new default [`ConductorConfig`] with environment path
 /// and keystore all in the same directory.
-pub fn create_config(environment_path: PathBuf, con_url: url2::Url2) -> ConductorConfig {
+pub fn create_config(environment_path: PathBuf, con_url: Option<url2::Url2>) -> ConductorConfig {
     let mut conductor_config = ConductorConfig {
         environment_path: environment_path.clone().into(),
         ..Default::default()
     };
     let mut keystore_path = environment_path;
     keystore_path.push("keystore");
-    conductor_config.keystore = KeystoreConfig::LairServer {
-        connection_url: con_url,
-    };
+    match con_url {
+        Some(url) => {
+            conductor_config.keystore = KeystoreConfig::LairServer {
+                connection_url: url,
+            };
+        }
+        None => {
+            conductor_config.keystore = KeystoreConfig::LairServerInProc { lair_root: None };
```

Comment:
Does the `None` here do what we want? Namely putting the lair database / configuration in the sandbox directory. Or does it default to the central system location?

Reply:
Yes and no does it do what we want. It defaults to `<environment_path>/keystore` which is entirely sensible BUT produces a very long path when run inside Nix (CI or locally). I'm now setting this explicitly to shorten the path otherwise Lair (mio) complains that the path is too long to be used as a unix socket.

(This is the problem I was having yesterday and trying to debug on CI)

Reply:
Ahh, yes @matthme ran into this in launcher... I believe in that case the solution was to create a symlink in /tmp to the path so it wasn't too long.

Reply:
That would work too but I'm fairly confident that this path is a fixed length. It's from the Nix root, then a fixed-width tempdir directory generated by the sandbox CLI then a fixed with suffix. So there shouldn't be any problems running this on different systems